src/Extensions/Logging/Internal/EventSource/ActivityEnhensementObserver.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Diagnostics;
+using Microsoft.Extensions.Options;
+using Microsoft.Omex.Extensions.Abstractions.Activities;
+using Microsoft.Omex.Extensions.Abstractions.Activities.Processing;
+
+namespace Microsoft.Omex.Extensions.Logging
+{
+	internal class ActivityEnhensementObserver : IActivityStartObserver
+	{
+		private readonly IOptions<OmexLoggingOptions> m_options;
+
+		public ActivityEnhensementObserver(IOptions<OmexLoggingOptions> options)
+		{
+			m_options = options;
+		}
+
+		public void OnStart(Activity activity, object? payload)
+		{
+			if (m_options.Value.AddObsoleteCorrelationToActivity)
+			{
+#pragma warning disable CS0618
```

Comment:
>pragma warning disable CS0618 [](start = 1, length = 29)

Could you add a comment clarifying why this needs to be suppressed?

Reply:
Added, thank you

Reply:
I don't see the comment

---
In reply to: [394721545](https://github.com/microsoft/Omex/pull/169#discussion_r394721545) [](ancestors = 394721545)

Reply:
I'm not sure why it's not displayed :) it actually there https://github.com/microsoft/Omex/blob/master/src/Extensions/Logging/Internal/EventSource/ActivityEnhancementObserver.cs